src/lib/core-extension-service.ts

```diff
@@ -79,14 +79,34 @@ export class CoreExtensionService {
 	 *
 	 * @returns The {@link TelemetryService} object exported by the Core Extension if available, else null.
 	 */
-	public static getTelemetryService(): TelemetryService|null {
+	public static _getTelemetryService(): CoreTelemetryService|null {
 		if (CoreExtensionService.initialized) {
 			return CoreExtensionService.telemetryService;
 		}
 		throw new Error(messages.error.coreExtensionServiceUninitialized);
 	}
 }
 
+export class TelemetryService {
```

Comment:
Moving `TelemetryService` into `core-extension-service.ts` allowed me to make it so all telemetry events have to go through this class instead of calling `CoreExtensionService._getTelemetryService()` directly.